Framework/Algorithms/src/GetAllEi.cpp

```diff
@@ -478,8 +479,8 @@ bool GetAllEi::peakGuess(const API::MatrixWorkspace_sptr &inputWS, size_t index,
   double xOfMax(0), dXmax(0);
   double Intensity(0);
 
-  auto X = inputWS->readX(index);
-  auto S = inputWS->readY(index);
+  const auto X = inputWS->x(index);
```

Comment:
Is there a reason why these are copied instead of just using a reference? Are they changed during use?